frontend/packages/dev-console/src/components/user-preferences/SecureRouteFields.tsx

```diff
@@ -20,33 +15,53 @@ const SecureRouteFields: React.FC = () => {
   const [isTLSTerminationOpen, setIsTLSTerminationOpen] = React.useState<boolean>(false);
   const [isInsecureTrafficOpen, setIsInsecureTrafficOpen] = React.useState<boolean>(false);
 
+  const TerminationTypes = {
+    edge: t('devconsole~Edge'),
+    passthrough: t('devconsole~Passthrough'),
+    reencrypt: t('devconsole~Re-encrypt'),
+  };
+
+  const InsecureTrafficTypes = {
+    None: t('devconsole~None'),
+    Allow: t('devconsole~Allow'),
+    Redirect: t('devconsole~Redirect'),
+  };
+
+  const PassthroughInsecureTrafficTypes = {
+    None: t('devconsole~None'),
+    Redirect: t('devconsole~Redirect'),
+  };
+
```

Comment:
Similar here. Please calculate `terminationOptions` and `insecureTrafficOptions` as "variables". You can also move this code into the useMemo blocks below.

And here as well: Code might be easier to understand if you calculate just one `insecureTrafficOptions`.